packages/language-server/src/lib/DiagnosticMap.ts

```diff
@@ -0,0 +1,28 @@
+import { Diagnostic } from 'vscode-languageserver'
+
+export class DiagnosticMap {
+  private _map = new Map<string, Diagnostic[]>()
+
+  constructor(uris: string[]) {
+    for (const uri of uris) {
+      this._map.set(uri, [])
+    }
+  }
+
+  add(fileUri: string, diagnostic: Diagnostic) {
+    let entry = this._map.get(fileUri)
+    if (typeof entry === 'undefined') {
+      entry = []
+      this._map.set(fileUri, entry)
+    }
+    entry.push(diagnostic)
+  }
```

Comment:
This feels a little wonky to me 🤔

Reply:
I wish we had `entry().or_default()` from Rust's maps in TS.
Open to suggestions on how to rewrite it, function should be able to handle both existing and not seen before `fileUri`.

Reply:
Would this not work?
```ts
  add(fileUri: string, diagnostic: Diagnostic) {
    const entry = this._map.get(fileUri) || []

    this._map.set(fileUri, entry)
    entry.push(diagnostic)
  }
```

Reply:
yep, that works, thank you